ee/localserver/krypto-ec-middleware.go

```diff
@@ -459,3 +477,197 @@ func (bhr *bufferedHttpResponse) WriteHeader(code int) {
 func (bhr *bufferedHttpResponse) Bytes() []byte {
 	return bhr.buf.Bytes()
 }
+
+func (e *kryptoEcMiddleware) presenceDetectionCallback(challengeBox *challenge.OuterChallenge) {
+	// extract cmd req from challenge box
+	ctx, cancel := context.WithTimeout(context.Background(), presencedetection.DetectionTimeout)
+	defer cancel()
+
+	var cmdReq v2CmdRequestType
+	if err := json.Unmarshal(challengeBox.RequestData(), &cmdReq); err != nil {
+		e.slogger.ErrorContext(ctx,
+			"error unmarshaling cmd request",
+			"err", err,
+		)
+		return
+	}
+
+	// figure out if we need to do presence detection
+	detectionIntervalStr, ok := cmdReq.Headers[kolidePresenceDetectionIntervalHeaderKey]
+	if !ok || len(detectionIntervalStr) == 0 {
+		return
+	}
+
+	// no presence detection requested
+	if len(detectionIntervalStr) == 0 || detectionIntervalStr[0] == "" {
+		return
+	}
+
+	// presence detection is not yet available on linux
+	if runtime.GOOS == "linux" {
+		return
+	}
+
+	detectionIntervalDuration, err := time.ParseDuration(detectionIntervalStr[0])
+	if err != nil {
+		return
+	}
+
+	reasonKey := kolidePresenceDetectionReasonMacosHeaderKey
+	if runtime.GOOS == "windows" {
+		reasonKey = kolidePresenceDetectionReasonWindowsHeaderKey
+	}
+
+	reason := "authenticate"
```

Comment:
Do we want a friendlier default? At least on Windows I think we'd want the reason string to be a full sentence. Maybe `Kolide is requesting authentication` or something?

Reply:
updated defaults, added comments